light/lightchain.go

```diff
@@ -143,6 +143,11 @@ func (lc *LightChain) Odr() OdrBackend {
 	return lc.odr
 }
 
+// HeaderChain returns the underlying header chain.
+func (lc *LightChain) HeaderChain() *core.HeaderChain {
```

Comment:
I check the APIs, it seems we only use it for `header` retrieval. So is it possible we change the interface to `ChainHeaderReader`, and define another interface like `ChainReader`. In this way we can use `LightChain` directly

Reply:
Good idea. Right now I just want to get the APIs to work and didn't want to change consensus engine for that.